dts/arm: Move i2c2 node inside stm32fxxx dtsi file

ydamigos opened the pull request:
SoC stm32f334x8 doesn't support I2C2 port. This
patch moves i2c2 node inside stm32f303 and
stm32f373 dtsi files.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos committed:
dts/arm: Move i2c2 node inside stm32fxxx dtsi file

SoC stm32f334x8 doesn't support I2C2 port. This
patch moves i2c2 node inside stm32f303 and
stm32f373 dtsi files.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos commented:
@DWagenk Could you also review it?

DWagenk commented:
Commit message and commit content differ:
> This patch moves i2c2 node inside stm32f303xc and stm32f373xc dtsi files

but they are moved into `stm32f303` and `stm32f373` dtsi files.

I'd vote on renaming `stm32f303` to `stm32f303Xc` (and adjusting the include in boards/arm/stm32f3_disco/stm32f3_disco.dts) since there's some device tree relevant differences between some of the stm32f303 SoCs. This doesn't seem to be the case for stm32f373 SoCs, so I'm a little unsure whether this should be renamed or not.

ydamigos commented:
>Commit message and commit content differ:
>
>    This patch moves i2c2 node inside stm32f303xc and stm32f373xc dtsi files
>
>but they are moved into stm32f303 and stm32f373 dtsi files.
>
>I'd vote on renaming stm32f303 to stm32f303Xc (and adjusting the include in boards/arm/stm32f3_disco/stm32f3_disco.dts) since there's some device tree relevant differences between some of the stm32f303 SoCs. This doesn't seem to be the case for stm32f373 SoCs, so I'm a little unsure whether this should be renamed or not.

I changed the commit message to much the content. I didn't create a specific dtsi file for stm32f303xc.

@erwango Should we use the same SoC dtsi file naming such as SoC naming in ext/hal/st/stm32cube/stm32fxxx/soc? It will make it more clear how to add new dtsi SoC files in DT.

DWagenk commented:
LGTM

I like the proposal of structuring stm .dtsi files like in ext/hal/st/stm32cube/stm32fxxx/soc. Would there also be a good place for writing down the "rule", so that people porting new boards would find it?

erwango commented:
> I'd vote on renaming stm32f303 to stm32f303Xc

Makes sense. Same naming has been used already for stm32f103Xy.

> Should we use the same SoC dtsi file naming such as SoC naming in ext/hal/st/stm32cube/stm32fxxx/soc?

It seems like a good idea a priori, but I'm not sure we should go for it. 
One point is that stm32cube naming of CSMIS files is not uniform. For instance in stm32f4 series, stm32f412cx is same as stm32f412zx, while stm32f410cx differs from stm32f410tx by 3 IPs.
So, referring to stm32cube filenaming won't help us to acheive a rule without exceptions.
Besides, unlike device tree, CMSIS  model do not use inheritance, which might lead us to other conflicts later on. 
Stm32cube based file naming rule won't prevent fine tuning discussions at review, so I prefer we keep existing way of working and its degree of freedom.

erwango reviewed (approved):
LGTM.
stm32f303Xc naming is optional as only one stm32f303 is present in Zephyr today.

ydamigos committed:
dts/arm/st: fix dts inclusion for stm32f334

stm32f334 is not an expansion of stm32f303, since
stm32f303 has more I2C, SPI ports than stm32f303.

Fix this by including stm32f3.dtsi in stm32f334.dtsi.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

ydamigos commented:
>stm32f303Xc naming is optional as only one stm32f303 is present in Zephyr today.

I will keep the naming as it is for now.

I also pushed a commit to fix the wrong inclusion of stm32f303.dtsi in stm32f334.dtsi.
